grpc/src/main/java/com/linecorp/armeria/server/grpc/UnframedGrpcErrorHandlerBuilder.java

```diff
@@ -0,0 +1,245 @@
+     * result in an {@link IllegalStateException}.
+     *
+     * @param messageTypes The message types to register
+     */
+    @SafeVarargs
+    public final UnframedGrpcErrorHandlerBuilder registerMarshalledMessageTypes(
+            Class<? extends Message>... messageTypes) {
+        requireNonNull(messageTypes, "messageTypes");
+        checkState(jsonMarshaller == null,
+                   "Cannot register custom messageTypes because a custom JSON marshaller has already been " +
+                   "set. Use the custom marshaller to register custom message types."
+        );
+
+        if (marshalledMessageTypes == null) {
+            marshalledMessageTypes = new ArrayList<>();
+        }
+        marshalledMessageTypes.addAll(Arrays.asList(messageTypes));
+        return this;
+    }
+
+    /**
+     * Registers custom message types to be marshalled by the error handler.
+     *
+     * <p>This method allows registering message instances for custom error responses.
+     * If a custom JSON marshaller has already been set, calling this method will
+     * result in an {@link IllegalStateException}.
+     *
+     * @param messageTypes The collection of message types to register
+     */
+    public UnframedGrpcErrorHandlerBuilder registerMarshalledMessageTypes(
+            Iterable<? extends Class<? extends Message>> messageTypes) {
+        requireNonNull(messageTypes, "messageTypes");
+        checkState(jsonMarshaller == null,
+                   "Cannot register the collection of messageTypes because a custom JSON marshaller has " +
+                   "already been set. Use the custom marshaller to register custom message types."
+        );
+        if (marshalledMessageTypes == null) {
+            marshalledMessageTypes = new ArrayList<>();
+        }
+        marshalledMessageTypes.addAll(ImmutableList.copyOf(messageTypes));
+        return this;
+    }
+
+    /**
+     * Returns a newly created {@link UnframedGrpcErrorHandler}.
+     *
+     * <p>This method constructs a new {@code UnframedGrpcErrorHandler} with the
+     * current configuration of this builder.
+     */
+    public UnframedGrpcErrorHandler build() {
+        if (jsonMarshaller == null) {
+            jsonMarshaller = UnframedGrpcErrorHandlers.ERROR_DETAILS_MARSHALLER;
+            MessageMarshaller.Builder builder = jsonMarshaller.toBuilder();
+
+            if (marshalledMessages != null) {
+                for (final Message message : marshalledMessages) {
+                    builder = builder.register(message);
+                }
+            }
+
+            if (marshalledMessageTypes != null) {
+                for (final Class<? extends Message> messageType : marshalledMessageTypes) {
+                    builder = builder.register(messageType);
+                }
+            }
+
+            jsonMarshaller = builder.build();
+        }
+        if (responseTypes == null) {
```

Comment:
For aesthetics' sake:
```suggestion
        }

        if (responseTypes == null) {
```